src/app/tracing/services/highlighting.service.ts

```diff
@@ -177,39 +177,86 @@ export class HighlightingService {
+    ): LegendDisplayEntry[] {
 
-        const ruleIdToIsCommonLinkRuleMap: Record<RuleId, boolean> = {};
-        this.getCommonLinkEntries(state).forEach(
-            commonLinkRule => ruleIdToIsCommonLinkRuleMap[commonLinkRule.id] = true
+        const stationRulesToDisplay = this.enabledStatHRules.filter(rule =>
+            rule.showInLegend &&
+            (activeHighlightings.stations[rule.id] || this.isCommonLinkRule(rule))
+        );
+        const deliveryRulesToDisplay = this.enabledDelHRules.filter(rule =>
+            rule.showInLegend && (activeHighlightings.deliveries[rule.id])
         );
 
-        return {
-            stations: this.enabledStatHRules.filter(rule =>
-                rule.showInLegend &&
-                (activeHighlightings.stations[rule.id] || ruleIdToIsCommonLinkRuleMap[rule.id])
-            ).map(rule =>
-                ({ label: rule.name, color: this.mapToColor(rule.color), shape: rule.shape })
-            ),
-            deliveries: this.enabledDelHRules.filter(rule =>
-                rule.showInLegend && (activeHighlightings.deliveries[rule.id])
-            ).map(rule =>
-                ({ label: rule.name, color: this.mapToColor(rule.color), linePattern: rule.linePattern })
+        return this.statHighlightingRules
+            .map(rule => this.stationRuleToDisplayEntry(rule))
+            .concat(this.delHighlightingRules
+                .map(rule => this.deliveryRuleToDisplayEntry(rule))
             )
-        };
+            .reduce(
+                (array: LegendDisplayEntry[], current) => {
+                    const index = array.findIndex((entry) => entry.name === current.name);
+                    if (index >= 0) {
+                        //Should use legend.with(index,{newObj}) once we update to a newer TS version.
+                        const newArray = array;
+                        newArray[index] = {
+                            name: current.name,
+                            stationColor: array[index]?.stationColor ?? current?.stationColor,
+                            deliveryColor: array[index]?.deliveryColor ?? current?.deliveryColor,
+                            shape: array[index]?.shape ?? current?.shape
+                        };
+                        return newArray;
+                    } else {
+                        return [...array,
+                            {
+                                name: current.name,
+                                stationColor: array[index]?.stationColor ?? current?.stationColor,
+                                deliveryColor: array[index]?.deliveryColor ?? current?.deliveryColor,
+                                shape: array[index]?.shape ?? current?.shape
+                            }
+                        ];
+                    }
+                },
+                []
+            )
+            .map(rule => stationRulesToDisplay.some(enabledRule => enabledRule.name === rule.name)
```

Comment:
This test assumes uniqueness within the rule names.
If you have an active and an inactive station rule with the same name, both are visualized in the legend.

Unfortunately, I guess this means you have to change your procedure.

Reply:
I'm in communication with Marion on how to handle duplicate names in highlighting rules.